semgrep-core/src/engine/Match_search_mode.ml

```diff
@@ -364,15 +364,21 @@ let rec filter_ranges env xs cond =
                    ( fk,
                      [
                        G.Arg (G.L (G.String (re_str, fk)) |> G.e);
-                       G.Arg (G.N (G.Id ((mvar, fk), fki)) |> G.e);
+                       G.Arg
```

Comment:
maybe add a comment before 'let e' = ...' giving the same of the code in concrete syntax, so 
re.match(str(...),...)

Reply:
It's hard to infer back from the AST what is the actual code we try to generate for Eval_generic.